### PR TITLE
Add support for the virtual machine power on

### LINKED ISSUE #36: Virtual Machine __refresh overwrites the data incorrectly.
The __refresh method incorrectly replaces self.data with the response without stripping off the DATA_FIELD before hand.

Here is the virtual machine data from the get_all method
```
<simplivity.resources.virtual_machines.VirtualMachine object at 0x7f7ba98fd160>
{   'app_aware_vm_status': 'UNKNOWN',
    'cluster_group_ids': <DATA REMOVED>,
    'compute_cluster_parent_hypervisor_object_id': <DATA REMOVED>,
    'compute_cluster_parent_name': <DATA REMOVED>,
    'created_at': '2020-03-17T22:24:08Z',
    'datastore_id': <DATA REMOVED>,
    'datastore_name': <DATA REMOVED>,
    'deleted_at': 'NA',
    'host_id': <DATA REMOVED>,
    'hypervisor_management_system': <DATA REMOVED>,
    'hypervisor_management_system_name': <DATA REMOVED>,
    'hypervisor_object_id': <DATA REMOVED>,
    'hypervisor_type': 'VSPHERE',
    'id': <DATA REMOVED>,
    'name': 'ubuntu-bionic-18.04-cloudimg-20200218',
    'omnistack_cluster_id': <DATA REMOVED>,
    'omnistack_cluster_name': '<DATA REMOVED>,
    'policy_id': <DATA REMOVED>,
    'policy_name': <DATA REMOVED>,
    'replica_set': [   {   'id': <DATA REMOVED>,
                           'role': 'SECONDARY'},
                       {   'id': <DATA REMOVED>,
                           'role': 'PRIMARY'}],
    'state': 'ALIVE'} 
```

Here is the virtual machine data after a refresh occurs
```
<simplivity.resources.virtual_machines.VirtualMachine object at 0x7f7ba98fd160>
{   'virtual_machine': {   'app_aware_vm_status': 'UNKNOWN',
                           'cluster_group_ids': <DATA REMOVED>,
                           'compute_cluster_parent_hypervisor_object_id': <DATA REMOVED>,
                           'compute_cluster_parent_name': <DATA REMOVED>,
                           'created_at': '2020-03-17T22:24:08Z',
                           'datastore_id': <DATA REMOVED>,
                           'datastore_name': <DATA REMOVED>,
                           'deleted_at': 'NA',
                           'ha_resynchronization_progress': 100,
                           'ha_status': 'SAFE',
                           'host_id': <DATA REMOVED>,
                           'host_name': <DATA REMOVED>,
                           'hypervisor_allocated_capacity': 12552624407,
                           'hypervisor_allocated_cpu': -1,
                           'hypervisor_consumed_cpu': 0,
                           'hypervisor_consumed_memory': 0,
                           'hypervisor_cpu_count': 1,
                           'hypervisor_folder_name': <DATA REMOVED>,
                           'hypervisor_free_space': 11460194304,
                           'hypervisor_instance_id': <DATA REMOVED>,
                           'hypervisor_is_template': False,
                           'hypervisor_managed_by': '',
                           'hypervisor_management_system': <DATA REMOVED>,
                           'hypervisor_management_system_name': <DATA REMOVED>,
                           'hypervisor_object_id': <DATA REMOVED>,
                           'hypervisor_total_memory': 1024,
                           'hypervisor_type': 'VSPHERE',
                           'hypervisor_uuid': <DATA REMOVED>,
                           'hypervisor_virtual_disk_count': 1,
                           'hypervisor_virtual_machine_power_state': 'OFF',
                           'id': <DATA REMOVED>,
                           'modified_at': 'Thu Jan 01 00:00:00 GMT 1970',
                           'name': 'ubuntu-bionic-18.04-cloudimg-20200218',
                           'omnistack_cluster_id': <DATA REMOVED>,
                           'omnistack_cluster_name': <DATA REMOVED>,
                           'policy_id': <DATA REMOVED>,
                           'policy_name': <DATA REMOVED>,
                           'replica_set': [   {   'id': <DATA REMOVED>,
                                                  'role': 'PRIMARY'},
                                              {   'id': <DATA REMOVED>,
                                                  'role': 'SECONDARY'}],
                           'state': 'ALIVE'}} 
```

### PR DESCRIPTION
**Pull request by KalpanaBhardwaj**
Signed-off-by: KalpanaBhardwaj <bhardwaj@hpe.com>

### Description
Add support for the virtual machine power on

### Issues Resolved
Fixes #36 

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass(`$ tox`).
- [x] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [x] New endpoints supported are updated in the endpoints-support.md file.
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: simplivity/resources/virtual_machines.py [hunk 1]
```diff
@@ -340,3 +340,20 @@ def power_off(self, timeout=-1):
...
         return self
+
+    def power_on(self, timeout=-1):
+        """Power on virtual machine.
+
+        Args:
+            timeout: Time out for the request in seconds.
+
+        Returns:
+            self: Returns the same object.
```
Comment (svt-kmcdaniel) [inline simplivity/resources/virtual_machines.py]: This should return true or false, nothing, right since we are editing the object in place?

you don't want the end user to have to replace their existing objects

File: tests/unit/resources/test_virtual_machines.py [hunk 2]
```diff
@@ -285,6 +285,18 @@ def test_power_off(self, mock_get, mock_post):
...
+    @mock.patch.object(Connection, "get")
+    def test_power_on(self, mock_get, mock_post):
+        mock_post.return_value = None, [{'object_id': '12345'}]
+        mock_get.return_value = {'virtual_machines': {'id': '12345'}}
+
+        vm1_data = {'name': 'name1', 'id': '12345'}
+        vm = self.machines.get_by_data(vm1_data)
+
+        vm.power_on()
+        mock_post.assert_called_once_with('/virtual_machines/12345/power_on', None, custom_headers={'Content-type': 'application/vnd.simplivity.v1.14+json'})
```
Comment (svt-kmcdaniel) [inline tests/unit/resources/test_virtual_machines.py:316]: check the data within vm to ensure the refresh worked

**Review by svt-kmcdaniel — CHANGES_REQUESTED** [review]

File: simplivity/resources/virtual_machines.py [hunk 3]
```diff
@@ -340,3 +340,20 @@ def power_off(self, timeout=-1):
...
+        """
+        method_url = "{}/{}/power_on".format(URL, self.data["id"])
+        custom_headers = {'Content-type': 'application/vnd.simplivity.v1.14+json'}
+
+        self._client.do_post(method_url, None, timeout, custom_headers)
+        self.__refresh()
```
Comment (svt-kmcdaniel) [inline simplivity/resources/virtual_machines.py:357]: FYI, refresh() is currently broken and reported in issue #36

File: simplivity/resources/virtual_machines.py (hunk 1 above)
Comment (KalpanaBhardwaj) [reply simplivity/resources/virtual_machines.py]: Will update as per suggestion

Comment (KalpanaBhardwaj) [reply simplivity/resources/virtual_machines.py]: Done

File: tests/unit/resources/test_virtual_machines.py (hunk 2 above)
Comment (KalpanaBhardwaj) [reply tests/unit/resources/test_virtual_machines.py:316]: Done

File: simplivity/resources/virtual_machines.py (hunk 3 above)
Comment (KalpanaBhardwaj) [reply simplivity/resources/virtual_machines.py:357]: I have fixed the __refresh() in this PR.

File: simplivity/resources/virtual_machines.py [hunk 4]
```diff
@@ -167,7 +167,7 @@ def __init__(self, connection, resource_client, data):
     def __refresh(self):
         """Updates the VM data."""
         resource_uri = "{}/{}".format(URL, self.data["id"])
-        self.data = self._client.do_get(resource_uri)
+        self.data = self._client.do_get(resource_uri)['virtual_machine']
```
Comment (svt-kmcdaniel) [inline simplivity/resources/virtual_machines.py:170]: This should be a separate commit, with and updated CHANGELOG.md update.

Comment (KalpanaBhardwaj) [reply simplivity/resources/virtual_machines.py:170]: Updated the CHANGELOG.md and also segregated the 2 commits, One for the fixing the issue and other to support the virtual machine power on

**Review by svt-kmcdaniel — APPROVED** [review]

**Review by sijeesh — APPROVED** [review]
